Applications/Spire/Source/KeyBindings/DestinationItemDelegate.cpp

```diff
@@ -0,0 +1,23 @@
+#include "Spire/KeyBindings/DestinationItemDelegate.hpp"
+#include "Spire/KeyBindings/InputFieldEditor.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+
+DestinationItemDelegate::DestinationItemDelegate(QWidget* parent)
+  : KeyBindingItemDelegate(parent) {}
+
+QWidget* DestinationItemDelegate::createEditor(QWidget* parent,
+    const QStyleOptionViewItem& option, const QModelIndex& index) const {
+  auto editor = new InputFieldEditor(index.data().value<QString>(),
+    {"One", "Two", "Three"}, static_cast<QWidget*>(this->parent()));
```

Comment:
I'm saying here that I don't want to use the parent that's given as a parameter, instead I want to use the parent that was set when the delegate was constructed. The QWidget's parent() function returns a QObject* instead of a QWidget*, so I make that cast.

Reply:
Good idea in general, luckily there is a method ```parentWidget()``` that solves this common issue.

Reply:
It looks like the parentWidget() function isn't available to delegates.